spiketoolkit/sortingcomponents/detection.py

```diff
@@ -116,11 +112,17 @@ def detect_spikes(recording, channel_ids=None, detect_threshold=5, n_pad_ms=2, u
     all_channel_times = [[] for ii in range(len(channel_ids))]
     all_channel_amps = [[] for ii in range(len(channel_ids))]
 
+    snippet_len = int(snippet_size_sec * recording.get_sampling_frequency())
+    reference_frames = np.linspace(snippet_len+1, recording.get_num_frames() - snippet_len,
+                                   n_snippets_for_threshold)
+    snippets = recording.get_snippets(reference_frames=reference_frames, snippet_len=snippet_len)
+    traces_mad = np.concatenate(snippets, 1)
+    thresholds = detect_threshold * np.median(np.abs(traces_mad) / 0.6745, 1)[:, None]
+
```

Comment:
This logic should be more explain in the doc. Maybe we could have other strategies than getting snippet linspace over time.